test/integration/interaction/Discovery.hpp

```diff
@@ -0,0 +1,77 @@
+#ifndef IN_TEST_DISCOVERY_HPP
+#define IN_TEST_DISCOVERY_HPP
+
+#include "BigHelloWorld.h"
+#include "Gateway.hpp"
+#include "Client.hpp"
+
+#include <uxr/client/client.h>
+#include <ucdr/microcdr.h>
+
+#include <gtest/gtest.h>
+#include <iostream>
+#include <thread>
+
+class Discovery
+{
+public:
+    Discovery(int transport, const std::vector<uint16_t>& agent_ports)
+    : agent_ports_(agent_ports)
+    , transport_(transport)
+    {
+    }
+
+    void unicast(const std::vector<uint16_t>& discovery_ports)
+    {
+        std::vector<uxrAgentAddress> agent_list(discovery_ports.size());
+        for(size_t i = 0; i < agent_list.size(); ++i)
```

Comment:
Maybe a `for` with `auto` is clearer.

Reply:
I refactored the code with a range-based loop, I think that because the element is not the iterator it is cool to know what is it in the loop definition (an `uint16_t`).

Reply:
Perfect!